src/barebones_mpc/nominal_path/abstract_NP.py

```diff
@@ -0,0 +1,82 @@
+# coding=utf-8
+
+from abc import ABC, ABCMeta, abstractmethod
+from typing import TypeVar, Union, Any, Type, Tuple, List, Dict
+import numpy as np
+from gym import wrappers as gym_wrappers
+from gym import make as gym_make
+
+from src.barebones_mpc.abstract_model_predictive_control_component import AbstractModelPredictiveControlComponent
+
+
+class AbstractNominalPath(ABC, AbstractModelPredictiveControlComponent):
+    def __init__(self, sample_length, input_dimension):
+        super().__init__()
+
+        self.sample_length = sample_length
+        self.input_dimension = input_dimension
+
+    @classmethod
+    def _specialized_config_key(cls) -> str:
+        return "nominal_path_bootstrap"
+
+    @classmethod
+    def _specialized_config_required_fields(cls) -> List[str]:
+        return []
+
+    def _config_pre__init__callback(
+        self, config: Dict, specialized_config: Dict, init__signature_values_from_config: Dict
+    ) -> Dict:
+        try:
+            horizon = config["hparam"]["sampler_hparam"]["horizon"]
+            time_step = config["hparam"]["sampler_hparam"]["steps_per_prediction"]
+        except KeyError as e:
+            raise KeyError(
+                f"{self.NAMED_ERR()} There's required baseclass parameters missing in the config file. Make sure that "
+                f"both following key exist: "
+                f"`hparam:sampler_hparam:horizon` and `hparam:sampler_hparam:steps_per_prediction`\n"
+                f"{e}"
+            ) from e
+
+        values_from_callback = {
+            "sample_length": int(horizon / time_step),
+            "input_dimension": config["environment"]["input_space"]["dim"],
+        }
+        return values_from_callback
+
+    def _config_post__init__callback(self, config: Dict) -> None:
+        pass
+
+    @abstractmethod
+    def bootstrap(self, state_t0=None) -> Any:
+        """ Bootstrap the initial nominal path
+
+        :param state_t0:
+        :return initial_nominal_path
+        """
+        pass
+
```

Comment:
As specified before, it would be great to be able to dynamically adjust certain parameters.